broadcastclients/broadcastclients.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2021-2022, Offchain Labs, Inc.
+// For license information, see https://github.com/nitro/blob/master/LICENSE
+
+package broadcastclients
+
+import (
+	"context"
+	"sync/atomic"
+
+	"github.com/ethereum/go-ethereum/log"
+
+	"github.com/offchainlabs/nitro/arbutil"
+	"github.com/offchainlabs/nitro/broadcastclient"
+	"github.com/offchainlabs/nitro/util/contracts"
+)
+
+type BroadcastClients struct {
+	clients []*broadcastclient.BroadcastClient
+
+	// Use atomic access
+	connected int32
+}
+
+func NewBroadcastClients(
+	config broadcastclient.Config,
+	l2ChainId uint64,
+	currentMessageCount arbutil.MessageIndex,
+	txStreamer broadcastclient.TransactionStreamerInterface,
+	fatalErrChan chan error,
+	bpVerifier contracts.BatchPosterVerifierInterface,
+) (*BroadcastClients, error) {
+	urlCount := len(config.URLs)
+	if urlCount <= 0 {
+		return nil, nil
+	}
+
+	clients := BroadcastClients{}
+	clients.clients = make([]*broadcastclient.BroadcastClient, 0, urlCount)
+	var lastClientErr error
+	for _, address := range config.URLs {
+		client, err := broadcastclient.NewBroadcastClient(
+			config,
+			address,
+			l2ChainId,
+			currentMessageCount,
+			txStreamer,
+			fatalErrChan,
+			bpVerifier,
+			func(delta int32) { clients.AdjustCount(delta) },
+		)
+		if err != nil {
+			lastClientErr = err
+			log.Warn("init broadcast client failed", "address", address)
+		}
+		clients.clients = append(clients.clients, client)
+	}
+	if len(clients.clients) == 0 {
+		log.Error("no connected feed on startup, last error: %w", lastClientErr)
+	}
+
+	return &clients, nil
+}
+
+func (bcs *BroadcastClients) AdjustCount(delta int32) {
+	atomic.AddInt32(&bcs.connected, delta)
+	if atomic.LoadInt32(&bcs.connected) <= 0 {
```

Comment:
`atomic.AddInt32` returns the new value, so we don't need to load it afterwards

Reply:
:+1: